### PR TITLE
Adds tests for graph.py

### PR DESCRIPTION
**Pull request by Fleker**
As part of #339, I started to write tests to verify existing behavior. During tests, I found some additional instructions to add to the README, and a fix to be made to the graph.py souce.

Let me know if the current changes look good, and if so I'll continue work on the issue.

### CODE REVIEW

File: utils/lib/rr_graph/tests/test_graph.py [hunk 1]
```diff
@@ -0,0 +1,772 @@
+#!/usr/bin/env python3
```
Comment (litghost) [inline utils/lib/rr_graph/tests/test_graph.py]: Any reason not to use (unittest)[https://docs.python.org/3/library/unittest.html] or equivilent?

**Comment by mithro** [comment]
Hi @Fleker,

Thank you for writing these tests! 

As Keith suggests we should use an existing testing framework. The unittest framework that is included with Python 3 is probably a good idea. `py.test` might be the other option.

Tim 'mithro' Ansell

File: utils/lib/rr_graph/tests/test_graph.py (hunk 1 above)
Comment (Fleker) [reply utils/lib/rr_graph/tests/test_graph.py]: I don't have a particular reason. I can definitely reformat the file, I wanted to make sure I wasn't using an unexpected framework that wasn't wanted in the project.

**Comment by mithro** [comment]
@Fleker, I have added a WIP title, feel free to remove once you have finished the reformat. 

Thanks once again!

**Comment by Fleker** [comment]
I changed the tests to use Python3's unittest. A second PR will add this and more tests to the Travis build.

File: utils/lib/rr_graph/tests/test_graph.py [hunk 2]
```diff
@@ -0,0 +1,754 @@
...
+        )
+        self.assertEqual(
+            P(x=0, y=0),
+            g.block_grid[P(0, 0)].position
+        )
+        self.assertEqual(
+            P(x=2, y=1),
+            g.block_grid[P(2, 1)].position
+        )
+        try:
```
Comment (mithro) [inline utils/lib/rr_graph/tests/test_graph.py]: `self.assertRaises`

Comment (Fleker) [reply utils/lib/rr_graph/tests/test_graph.py]: Updated.

**Comment by mithro** [comment]
Thank you for your pull request!

**Pull request merged**